client/go/outline/iptable/ip_table.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptable
+
+import (
+	"fmt"
+	"net/netip"
+)
+
+type IPTable[D any] interface {
+	AddPrefix(prefix netip.Prefix, dialer D) error
+	Lookup(ip netip.Addr) (D, error)
+}
```

Comment:
```suggestion
type IPTable[V any] interface {
	AddPrefix(prefix netip.Prefix, value V) error
	Lookup(ip netip.Addr) (V, error)
}
```

We don't need to think of the value as a dialer here.